src/Magnum/Platform/EmscriptenApplication.h

```diff
@@ -0,0 +1,1509 @@
+        }
+
+        /** @brief Key name */
+        std::string keyName() const {
+            if((_event->key[0] >= 'a' && _event->key[0] <= 'z') ||
+               (_event->key[0] >= 'A' && _event->key[0] <= 'Z')) {
+                return _event->key;
+            }
+            return _event->code;
+        }
+
+        /** @brief Modifiers */
+        Modifiers modifiers() const {
+            Modifiers m;
+            if(_event->ctrlKey) m |= Modifier::Ctrl;
+            if(_event->shiftKey) m |= Modifier::Shift;
+            if(_event->altKey) m |= Modifier::Alt;
+            if(_event->metaKey) m |= Modifier::Super;
+            return m;
+        }
+
+    private:
+        explicit KeyEvent(const EmscriptenKeyboardEvent* event): _event(event) {}
+
+        const EmscriptenKeyboardEvent* _event;
+
+        /* Translate emscripten key code (as defined by
+           https://www.w3.org/TR/uievents-code/#key-code-attribute-value)
+           to Key enum.
+
+            @param key Keyboard layout dependent key string, e.g. 'a', or '-'
+            @param code Keyboard layout independent key string, e.g. 'KeyA' or 'Minus'.
+                        Note that the y key on some layouts may result in 'KeyZ'.
+         */
+        static Key toKey(const EM_UTF8* key, const EM_UTF8* code);
+};
+
+CORRADE_ENUMSET_OPERATORS(EmscriptenApplication::MouseMoveEvent::Buttons)
+
+/**
+@brief Text input event
+
+@see @ref TextEditingEvent, @ref textInputEvent()
+*/
+class EmscriptenApplication::TextInputEvent {
+    friend EmscriptenApplication;
+
+    public:
+        /** @brief Copying is not allowed */
+        TextInputEvent(const TextInputEvent&) = delete;
+
+        /** @brief Moving is not allowed */
+        TextInputEvent(TextInputEvent&&) = delete;
+
+        /** @brief Copying is not allowed */
+        TextInputEvent& operator=(const TextInputEvent&) = delete;
+
+        /** @brief Moving is not allowed */
+        TextInputEvent& operator=(TextInputEvent&&) = delete;
+
+        /** @brief Whether the event is accepted */
+        bool isAccepted() const { return _accepted; }
+
+        /**
+         * @brief Set event as accepted
+         *
+         * If the event is ignored (i.e., not set as accepted), it might be
+         * propagated elsewhere, for example to another screen when using
+         * @ref BasicScreenedApplication "ScreenedApplication". By default is
+         * each event ignored and thus propagated.
+         */
+        void setAccepted(bool accepted = true) { _accepted = accepted; }
+
+        /** @brief Input text in UTF-8 */
+        Containers::ArrayView<const char> text() const { return _text; }
+
+    private:
+        explicit TextInputEvent(Containers::ArrayView<const char> text): _text{text}, _accepted{false} {}
+
+        const Containers::ArrayView<const char> _text;
+        bool _accepted;
+};
+
+/**
+@brief Text editing event
+
+@see @ref textEditingEvent()
+*/
+class EmscriptenApplication::TextEditingEvent {
```

Comment:
Since the EmscriptenApp doesn't implement any of this, I think this shouldn't be here at all. The same is with GLFW, it has no `textEditingEvent()` at all.